spec/services/mentions/create_all_spec.rb

```diff
@@ -1,88 +1,140 @@
-    described_class.call(comment)
+    user3 = create(:user)
+
+    set_markdown_and_save(notifiable, "Hello @#{user.username} @#{user3.username}, you are cool.")
+    described_class.call(notifiable)
     expect(Mention.all.size).to eq(2)
   end
 
   it "deletes one of multiple mentions if one of multiple is deleted" do
-    user2 = create(:user)
-    comment.body_markdown = "Hello @#{user.username} @#{user2.username}, you are cool."
-    comment.save
-    described_class.call(comment)
+    user3 = create(:user)
+
+    set_markdown_and_save(notifiable, "Hello @#{user.username} @#{user3.username}, you are cool.")
+    described_class.call(notifiable)
     expect(Mention.all.size).to eq(2)
-    comment.body_markdown = "Hello @#{user2.username}, you are cool."
-    comment.save
-    described_class.call(comment)
+
+    set_markdown_and_save(notifiable, "Hello @#{user3.username}, you are cool.")
+    described_class.call(notifiable)
     expect(Mention.all.size).to eq(1)
   end
 
-  it "creates a mention on creation of comment" do
-    described_class.call(comment2)
+  it "creates a mention on creation of notifiable" do
+    described_class.call(notifiable)
     expect(Mention.all.size).to eq(1)
   end
 
-  it "does not create a mention without valid mentionable" do
-    comment2.update_column(:body_markdown, "")
-    described_class.call(comment2)
+  it "does not create a mention when the user mentions themselves" do
+    set_markdown_and_save(notifiable, "Me, Myself and I @#{user2.username}")
+    described_class.call(notifiable)
     expect(Mention.all.size).to eq(0)
   end
+end
 
-  it "does not create a mention when the user mentions themselves" do
-    comment.body_markdown = "Me, Myself and I @#{user2.username}"
-    comment.save
-    described_class.call(comment)
+RSpec.shared_examples "valid notifiable with embedded mentions" do
+  it "does not create a mention" do
+    comment = create(:comment, user_id: user2.id, commentable: article, body_markdown: "Hi there, @#{user.username}")
+    liquid_tag_markdown = "Check out this comment: {% comment #{comment.id_code_generated} %}"
+
+    notifiable.update_column(:body_markdown, liquid_tag_markdown)
+    described_class.call(notifiable)
     expect(Mention.all.size).to eq(0)
   end
 end
+
+RSpec.shared_examples "invalid notifiable and has mentions" do
+  it "does not create a mention without valid mentionable" do
+    notifiable.update_column(:body_markdown, "")
+    described_class.call(notifiable)
+    expect(Mention.all.size).to eq(0)
+  end
+end
+
+RSpec.describe Mentions::CreateAll, type: :service do
+  let(:user)               { create(:user) }
+  let(:user2)              { create(:user) }
+  let(:article)            { create(:article, user_id: user.id) }
+  let(:markdown)           { "Hello, you are cool." }
+  let(:mention_markdown)   { "Hello @#{user.username}, you are cool." }
+
+  def set_markdown_and_save(notifiable, markdown)
+    notifiable.body_markdown = markdown
```

Comment:
Is there a reason for not using `update` here?

```ruby
notifiable.update(body_markdown: markdown)
```

Reply:
No reason -- I just pulled this into a method while refactoring this into shared examples and didn't think to do it, but that's a great call 👍🏽